runtime/include/chplmemtrack.h

```diff
@@ -29,6 +29,8 @@
 #include <stdlib.h>
 
 
+///// These entry points support debugging.
+
```

Comment:
The main difference I wanted to express is that one set is required to support the functionality provided by the MemTracking module, while the other set is not (at least not directly).

I wanted to point out that interfaces in the second set should *not* be considered candidates for dead-code removal even if they were not bound by the average Chapel program.  The intent behind them is to be available for debugging purposes (if not bound specifically through extern declarations in the Chapel code or in special C code linked to the Chapel program).

I would be happy to take your recommendations on how to say this succinctly in the comments.

THH
________________________________
From: Greg Titus [notifications@github.com]
Sent: Thursday, May 21, 2015 10:47 AM
To: chapel-lang/chapel
Cc: Tom Hildebrandt
Subject: Re: [chapel] Add memLeaksByDesc config const to MemTracking.chpl (#1983)


In runtime/include/chplmemtrack.h<https://github.com/chapel-lang/chapel/pull/1983#discussion_r30827319>:

> @@ -29,6 +29,8 @@
>  #include <stdlib.h>
>
>
> +///// These entry points support debugging.
> +


I think this and the comment below are a mis-characterization of the difference between the two sets of functions. These ones you describe here as supporting debugging are for more than just that. The programmer might have other reasons for being interested about memory usage, either during or at the end of execution. If you really need to to split the functions into two groups, you could say this first set is to support the user interface embodied in the module code, while the second set supports the data collection upon which the first set relies. I would argue that chpl_reportMemInfo() belongs in the first set, though, since its only purpose is to cause the reporting requested by the various config consts.

—
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/1983/files#r30827319>.